base/boot.jl

```diff
@@ -218,19 +218,12 @@ end
 
 abstract DirectIndexString <: AbstractString
 
-immutable ASCIIString <: DirectIndexString
+immutable String <: AbstractString
     data::Array{UInt8,1}
-    ASCIIString(d::Array{UInt8,1}) = new(d)
+    String(d::Array{UInt8,1}) = new(d)
```

Comment:
Because I'm trying to get this PR merged before someone introduces more conflicts and/or breaks the tests. You have no idea how many times I've rebased this and how often it breaks when someone changes something seemingly unrelated.